### PR TITLE
ARROW-2847: [Packaging] Fix artifact name matching for conda forge packages

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

File: dev/tasks/crossbow.py [hunk 1]
```diff
@@ -230,8 +230,26 @@ def github_assets(self, task):
...
+
+        artifacts = {}
+        for artifact in task.artifacts:
+            # artifact can be a regex pattern
+            pattern = re.compile(artifact)
+            matches = list(filter(None, map(pattern.match, assets.keys())))
+            num_matches = len(matches)
+
+            # validate artifact pattern matches single asset
+            if num_matches > 1:
```
Comment (kszucs) [inline dev/tasks/crossbow.py]: @cpcloud This is the main downside of using patterns.

Comment (cpcloud) [reply dev/tasks/crossbow.py]: Does this need to handle the case of `num_matches == 0`? Shouldn't there be at least 1 match?

**Comment by kszucs** [comment]
@cpcloud the only failing build is `centos-7`, created a ticket for it: https://issues.apache.org/jira/browse/ARROW-2868

File: dev/tasks/crossbow.py (hunk 1 above)
Comment (kszucs) [reply dev/tasks/crossbow.py]: We highlight artifacts with `num_matches == 0` as `MISSING`.

**Comment by kszucs** [comment]
[asciicast](https://asciinema.org/a/192205)

@cpcloud The missing `conda-win` packages are resolved by https://github.com/apache/arrow/pull/2274/commits/333bae638787c2c28aa0c77be46423d3908ee26e , just need to resubmit the tasks.

**Review by cpcloud — APPROVED** [review]
+1 LGTM

**Comment by cpcloud** [comment]
Nice output!

**Comment by cpcloud** [comment]
Merging.

**Comment by cpcloud** [comment]
Thanks @kszucs!

**Comment by wesm** [comment]
Thanks @kszucs! Hoo boy do we ever need some mocks and unit tests, after 0.10.0 I guess

**Pull request closed without merging**